modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAngularClientCodegen.java

```diff
@@ -360,6 +372,18 @@ public Boolean getStringEnums() {
         return stringEnums;
     }
 
+    public boolean getQueryParamObjectFormatDot() {
+        return queryParamObjectFormat != null && queryParamObjectFormat.equals("dot");
```

Comment:
this can be simplified a bit but swapping operands:
```suggestion
        return dot.equals(queryParamObjectFormat);
```

however, switching from `string` to `enum` would eliminate the problem altogether